lib/models/project.js

```diff
@@ -75,14 +76,38 @@ class Project {
     instrumentation.project = this;
 
     this.emberCLIVersion = emberCLIVersion;
+
     this._nodeModulesPath = null;
+
+    if (this.cli && this.cli._packageInfoCache) {
```

Comment:
* If somebody directly creates a project, they could pass in null for cli. I also saw it in a piece of code checking for isEmberCliProject. It is admittedly not consistent across the file.
* Originally MockCLI didn't have a _packageInfoCache entry. I changed the code the other day to add one. It was tripping over that issue in tests a long time ago, so the second test really isn't needed any more. First one, not so sure.

Reply:
Removed the second part of the test.

Reply:
> If somebody directly creates a project, they could pass in null for cli.

should this be allowed? is it because of our mocks in tests?

Reply:
Probably not, but I found that without the second part of the test (I have re-added it), blueprint tests fail. Apparently they are somehow creating a project without a CLI, or an empty one. I fixed MockCLI a while back so it does create a _packageInfoCache, but it may not always be being used.